src/main.rs

```diff
@@ -112,9 +120,16 @@ async fn main() -> Result<(), Error> {
     let (task_tracker, error_rx) = TaskTracker::new();
 
     //let (chains, currencies) = rpc::prepare(config.chain, config.account_lifetime, config.depth).await
-    let currencies = HashMap::new();
 
     let rpc = env::var("KALATORI_RPC").unwrap();
+    let mut currencies = HashMap::new();
+    if let Ok(client) = WsClientBuilder::default().build(rpc.clone()).await {
```

Comment:
ugly way to fetch currencies

Reply:
No, wait, these currencies should come from config file. Then the daemon should start complaining if the network does not allow these upon connect and try to rotate endpoints.

Reply:
hmm, I think we should provide currencies that are supported on chain at the moment disregard the config. I mean if we have unsupported currencies in configuration, we will tell the frontend that we are able to receive those. 
@Slesarew wdyt?

Reply:
Right now it is supposed to report only currencies that were actually fetched from chain (and chain connection fails beforehand if they are not present). Thus we filter by both now. We could implement crash instead, but it is quite often that the chain fails to respond correctly, thus reconnections fix the issue eventually. There must be thus some empirical rule to stop trying, but it seems more of an improvement that a neccesity now.